### ISSUE TITLE
Question: Building with VS2019

### ISSUE THREAD

**Issue by Gnbrkm41** [xunit/xunit#1972]
It does look like at the moment the project only supports building with VS2017; is there any future plans to support VS2019 as well?
At the moment, trying to build the main `xunit` repo with VS2019 toolsets fail with `xunit.runner.utility\xunit.runner.utility.csproj : error MSB4057: The target "_GetRestoreSettingsPerFramework" does not exist in the project.` ([Which I'm not certain if I'm still missing an SDK](https://github.com/xunit/xunit/issues/1819) since I think I installed 10240 SDK correctly), and trying to build `xunit.analyzers` repo fails with `Unexpected MSBUILD version 16.200.19.27403. Please ensure MSBUILD.EXE v15 is on the path`. 

On a sidenote, after bypassing the check by editing the script, the build fails with package version conflict for Microsoft.CodeAnalysis.CSharp:
```cs
Errors in C:\users\gotos\source\repos\xunit.analyzers\tools\DocBuilder\DocBuilder.csproj
    Version conflict detected for Microsoft.CodeAnalysis.CSharp.
     Xunit.Analyzers.DocBuilder (>= 1.0.0) -> Microsoft.NETCore.App (>= 1.0.0) -> Microsoft.CodeAnalysis.CSharp (>= 1.3.0)
     Xunit.Analyzers.DocBuilder (>= 1.0.0) -> xunit.analyzers (>= 1.0.0) -> Microsoft.CodeAnalysis.CSharp.Workspaces (>= 1.2.2) -> Microsoft.CodeAnalysis.CSharp (>= 1.2.2).
```

**Comment by bradwilson** [comment]
For 3.0, the primary dev environment will be VS Code on Linux: https://github.com/xunit/xunit/issues/1733

I will attempt to make Visual Studio work when this transition is done (and I would be focused on the latest version). In the meantime, using VS 2017 is apparently going to be required.

**Comment by Gnbrkm41** [comment]
I see. Will look forward to the new version!

**Comment by KirillOsenkov** [comment]
I also can't build 2.4.1 due to the `The target "_GetRestoreSettingsPerFramework" does not exist in the project.` error. I doubt it's related to VS 2017 vs 2019, something else is going on. 

I have filed an issue against MSBuild https://github.com/microsoft/msbuild/issues/5473 to investigate this further. Once we know what's going on I'll reply here.

**Comment by KirillOsenkov** [comment]
OK looks like this might be the solution:
https://github.com/xunit/xunit/issues/1819

**Comment by KirillOsenkov** [comment]
I have a fix available: #2115

The problem that caused The target "_GetRestoreSettingsPerFramework" does not exist in the project. was this MSBuild.Sdk.Extras issue: novotnyllc/MSBuildSdkExtras#127 which was fixed in 1.6.61.
